⚡ [Meteor-Vite v4] Added lightning fast server-side HMR though Vite v6's new Environments API

JorgenVatle opened the pull request:
Vite v6 is bringing some exciting new changes for framework developers. Most notably in the context of Meteor is their Environments API, loosening some of the assumptions Vite would make in terms of the environment its running in.

Up until now, we've had Vite run as a separate process independently of Meteor to work around compatibility issues that come with Meteor's runtime environment. Running these as separate processes has involved quite a lot of work to ensure things are working somewhat reliably. 

What has taken thousands of lines of code to make the two work together can now be done in less than a hundred. And that's with HMR not only for the client, but now for the Meteor server as well. 🔥

## Installation
```sh
# Replace `vite-bundler` Atmosphere package with `vite`
meteor remove jorgenvatle:vite-bundler
meteor add jorgenvatle:vite@1.1.0-alpha.0

# Install Vite v6 and MeteorVite v3
meteor npm i vite@6 meteor-vite@alpha

# Append `--force` to the above command if you're using plugins that don't yet have a pre-release for Vite v6
# Find the latest release of jorgenvatle:vite by calling `meteor show jorgenvatle:vite --show-all`
```
MeteorVite's plugin configuration remains the same, so the above is all that's necessary to try out the new server-side HMR. 

## Key changes
- Vite can now take over the entire build process for the your Meteor server code. This is an improved iteration of the `serverEntry` option introduced in [`jorgenvatle:vite-bundler@2.1.0`](https://github.com/JorgenVatle/meteor-vite/releases/tag/vite-bundler%402.1.2)
  - It does not require any intermediary pre-bundling step and is done at runtime.
  -  Changes you make to server code benefit from the same near-instant HMR that you get on the client. 
  - Your Meteor server no longer needs to restart in-between changes you make changes to it.
- Vite now runs as a part of the Meteor runtime. 
  - HMR and assets are served through Meteor's own web server, dropping the need to manage more than one host in development. Should make things a lot easier to deal with if you need to expose the dev server to the local network, particularly for Cordova.
- In production, the same changes from https://github.com/JorgenVatle/meteor-vite/pull/173 apply. Your assets are built and served as static files without the need for pushing them through Meteor's build system.
- Client assets benefit from code splitting. What's not immediately required on first page load will be prefetched quietly in the background by the browser.


## Compatibility 
- ~~Vite v6 is still in alpha and there's not a lot of plugins that have been updated to match the new release. From my testing, Vite plugins for Vite v5 seems to work fine, but there's also a chance they might not. `@vitejs/plugin-vue@5` hasn't given me any issues this far.~~
- Since Vite v5, support for Node v14 has been dropped. As this depends on Vite v6, support for Meteor v2 is no longer possible. 
- Building the Meteor server with Vite is still in its experimental phase. To provide HMR for the server, modules with side-effects need to be cleaned up between changes. Currently we clear out your method and publication handlers between updates to prevent `Mongo.Collection` from yelling at you for defining/instantiating them twice. But there likely is a list of other things that might be other things that need to be cleaned up between reloads. Looking forward to any feedback on this.
  - If you run into issues, you can always just comment out the `serverEntry` field in your Vite config and it should behave just like previous versions of Meteor Vite.


## Progress
- [x] Build Meteor server with Vite in development
  - [x] HMR
  - [x] Test SSR
- [x] Refactor build process and remove redundant code from previous versions.
- [x] Build Meteor server with Vite in production
- [x] Inject React preamble automatically into app markup instead of requiring users to configure this manually.
- [ ] Add Meteor v2/Vite v4 compatibility flag for `meteor-vite` npm package. (Unless it's easier to maintain a separate release branch)
  - [ ] Enable compatibility mode based on currently installed Vite version.
- [x] Update example apps
  - [x] [Vue](https://github.com/JorgenVatle/meteor-vite/tree/v3.1-vite-vms/examples/vue)
  - [x] React
  - [x] Svelte
    - [x] Replace `rdb:svelte-meteor-data` with an implementation compatible with Svelte v5
  - [x] Solid

JorgenVatle committed:
Add useNode and wrapAsync to known module method names

JorgenVatle committed:
Use environment name as part of stubs base dir

JorgenVatle committed:
Fix issue where stubs would attempt to import directly from a guessed main module path (e.g. ostrio:cookies/main.js) instead of importing directly from the actually requested import path (ostrio:cookies).

JorgenVatle committed:
Remove startup wrapper for initial methods and publications

JorgenVatle committed:
Use Meteor startup block for initializing Vite dev server

JorgenVatle committed:
Use intermediary Meteor project when preparing packages for export analyzer

JorgenVatle committed:
Rely on separate package export analyzer project for omitting incompatible plugins from the project during build

JorgenVatle committed:
Add todo entry for handling packages that use both implicit and explicit exports

JorgenVatle committed:
Update lockfile

JorgenVatle committed:
Update directory nesting for package analyzer in/out paths

JorgenVatle committed:
Update code style preferences for project

JorgenVatle committed:
Add excluded library mappings

JorgenVatle committed:
Refactor directory structure for vite plugin entries

JorgenVatle committed:
Correct path to entry modules

JorgenVatle committed:
Move plugin entrypoint into dedicated subdirectory

JorgenVatle committed:
Move internal bootstrapper to entry directory

JorgenVatle committed:
Use subdirectories for bootstrap and plugin entries

JorgenVatle committed:
Move meteor directory into plugin entry

JorgenVatle committed:
Move plugin modules into plugin entry directory

JorgenVatle committed:
Move plugin settings module into entry directory

JorgenVatle committed:
Move ViteLoadRequest into plugin entry directory

JorgenVatle committed:
Move client modules into entry directory

JorgenVatle committed:
Move bootstrap index module into scripts subdir

JorgenVatle committed:
Move bootstrap scripts into entry directory

JorgenVatle committed:
Move production environment bootstrapper into entry dir

JorgenVatle committed:
Move config resolver into bootstrap lib

JorgenVatle committed:
Move dev server HMR into entry directory

JorgenVatle committed:
Move instance utility into entry directory

JorgenVatle committed:
Move dev server script into dedicated bootstrap entry module

JorgenVatle committed:
Rename runtime HMR module entry

JorgenVatle committed:
Move boilerplate into bootstrap directory

JorgenVatle committed:
Fix entry paths for meteor-vite package

JorgenVatle committed:
Update entry for bootstrap scripts

JorgenVatle committed:
Add type path declarations for package

JorgenVatle committed:
Remove typesVersions

JorgenVatle committed:
Correct path reference for meteor-vite

JorgenVatle committed:
Add dev server init to bootstrapper scripts

JorgenVatle committed:
Correct entry path for build plugin

JorgenVatle committed:
Correct entries for vite package

JorgenVatle committed:
Correct dist paths for build plugin and runtime

JorgenVatle committed:
Correct import string for bootstrappers

JorgenVatle committed:
Correct import path for package.json

JorgenVatle committed:
Revert "Fix issue where stubs would attempt to import directly from a guessed main module path (e.g. ostrio:cookies/main.js) instead of importing directly from the actually requested import path (ostrio:cookies)."

This reverts commit d4e9e994269473f0ea7726b38b333a3e8f7a8957.

JorgenVatle committed:
docs(changeset): Wrap dev server bootstrapper around Meteor.startup hook. Workaround for issue where changes to Meteor.settings.public aren't sent to clients. Ref: https://github.com/meteor/meteor/issues/13489

JorgenVatle committed:
Add MethodMap entries for useNode and wrapAsync

JorgenVatle committed:
Use slightly stricter types for UseNode and WrapAsync

JorgenVatle committed:
Merge pull request #250 from JorgenVatle/v3.1-vite-vms--reformat-dir-structure

V3.1 vite vms  reformat directory structure

JorgenVatle committed:
Add accounts-base to dependencies

JorgenVatle committed:
Revert "Revert "Fix issue where stubs would attempt to import directly from a guessed main module path (e.g. ostrio:cookies/main.js) instead of importing directly from the actually requested import path (ostrio:cookies).""

This reverts commit effc12e76da634186a81dab1a47851110f1e8a04.

JorgenVatle committed:
Add mainModule path to stub template args

JorgenVatle committed:
Use mainModulePath as default submodule, only overriding on explicit import path

JorgenVatle committed:
Use mainModule if no import path is provided

JorgenVatle committed:
Re-enable standard minifier packages

JorgenVatle committed:
Use input directory, not project directory for package analyzer builds

JorgenVatle committed:
Install missing packages

JorgenVatle committed:
Revert "Omit unsupported dependencies"

This reverts commit a3bcab1c9bace204499a9eda69c88e85a2de85ba.

JorgenVatle committed:
Revert "Disable incompatible minifier packages"

This reverts commit ce276a599db62b4f7edadc327f7e589c7469a70e.

JorgenVatle committed:
Rename meteor-v3-vue example dir to vue-zodern-relay

JorgenVatle committed:
Add option to control Vite plugin debug through environment variables

JorgenVatle committed:
Update environment key for debug option

JorgenVatle committed:
Add debug build logger

JorgenVatle committed:
Use BuildLogger for export analyzer logs

JorgenVatle committed:
Use BuildLogger for export analyzer logs

JorgenVatle committed:
Copy caches when creating export analyzer build

JorgenVatle committed:
Copy cache directories

JorgenVatle committed:
re-order path list

JorgenVatle committed:
Use environment variable rather than copy for package dirs inclusion

JorgenVatle committed:
Use shared directory for package analyzer in/out

timsun28 commented:
This is looking really promising and a great improvement to the meteor ecosystem! 
Is there anything I can help you with in terms of testing or creating sample projects? 

Update: Just saw this on Twitter: https://vite.dev/blog/announcing-vite6.html 🎉

JorgenVatle committed:
Merge package dirs with current environment

JorgenVatle committed:
Fix package dirs merging

JorgenVatle committed:
Add missing packages entry to files to copy

JorgenVatle committed:
Update log format for removed lines

JorgenVatle committed:
Use cyan for removed line content

JorgenVatle committed:
Add colorization helpers to use consistent colors for highlighted log items

JorgenVatle committed:
Use colorization helper for filepath highlighting

JorgenVatle committed:
Use standardized colorizers for arch logs

JorgenVatle committed:
Use build logger success message for package build time

JorgenVatle committed:
Do not copy local caches when preparing packages for export analyzer

JorgenVatle committed:
Move Colorize into a dedicated formatting module

JorgenVatle committed:
Scope package analyzer in/out dirs to the basename of the current project

JorgenVatle committed:
Use null instead of undefined for mainModulePath

JorgenVatle committed:
Store mainModulePath as null instead of undefined.

JorgenVatle committed:
Add parser for main module path (Meteor v3)

JorgenVatle committed:
Yield mainModulePath in v3 parser

JorgenVatle committed:
Ensure mainModulePath from package scope takes precedence when parsing packages

JorgenVatle committed:
Add rdb:svelte-meteor-data to bundle mocks

JorgenVatle committed:
Prevent atmosphere object merge from overriding mainModule paths from package scope

JorgenVatle committed:
Allow null type for mainModulePath

JorgenVatle committed:
Use absolute path for cache base dir

JorgenVatle committed:
Default mainModulePath to use path relative to the package root

JorgenVatle committed:
Rename absoluteMainModulePath property

JorgenVatle commented:
> Update: Just saw this on Twitter: https://vite.dev/blog/announcing-vite6.html 🎉

Oh, that's great news! 🙌

@timsun28 Thank you for offering to lend a hand. If you have the time, I think we could certainly expand the dependencies of the existing example projects or add some more projects using some of the more commonly used Meteor packages. They're very valuable for catching compatibility issues early. If there's any Meteor packages you've had issues with in the past, putting them in an example app would help a ton. 😄

JorgenVatle committed:
Use getModule helper instead of manual module list matching for mainModule getter

JorgenVatle committed:
Add paths for mockfiles to load request cache

JorgenVatle committed:
Add helper method for creating mocks from Meteor packages

JorgenVatle committed:
Store mock files along debug snippets

JorgenVatle committed:
Create parent directory for mocks

JorgenVatle committed:
Fix directory nesting for cache/debug files

JorgenVatle committed:
Correct splitting for cache basename

JorgenVatle committed:
Update directory structure for mocks

JorgenVatle committed:
Rename filename for bundle source

JorgenVatle committed:
Revert back to using absolute mainModulePaths by default

JorgenVatle committed:
Do main module path formatting in mainModule getter

JorgenVatle committed:
Perform import path formatting in getModule helper

JorgenVatle committed:
Resolve absolute path to package dirs env variable

JorgenVatle committed:
Correct import path for MeteorVite HMR module

JorgenVatle committed:
Add isopack path to package analyzer settings

JorgenVatle committed:
Use isopackPath from temporary Meteor project during build

JorgenVatle committed:
Store isopack manifest alongside other debug modules

JorgenVatle committed:
Improve formatting for isopack manifest

JorgenVatle committed:
Rename main module path property to avoid confusion with eagerly loaded main module

JorgenVatle committed:
Add lazy main module path to MeteorPackage

JorgenVatle committed:
Use lazy main module path when parsing and templating Meteor package

JorgenVatle committed:
Add fallback to lazy main module path in MeteorPackage mainModule getter

JorgenVatle committed:
Remove redundant mainModule parameter in serializer method

JorgenVatle committed:
Remove redundant mainModule parameter from stubTemplate

JorgenVatle committed:
Include import path in stub template comments

JorgenVatle committed:
Fix issue where stubs would try to import directly from package's main module instead of the actually requested module path

JorgenVatle committed:
Remove redundant try-catch block for stub templates

JorgenVatle committed:
Refactor main module path determination into PackageSubmodule.ts

JorgenVatle committed:
Update changelog entry to include note about changes to package parser

JorgenVatle committed:
Remove Vite version overrides

JorgenVatle committed:
Install vite v6

JorgenVatle committed:
Remove vite from repository-root dependencies

JorgenVatle committed:
Upgrade to Vite v6

JorgenVatle committed:
Update lockfile for meteor-vite

JorgenVatle committed:
Re-add vite as repository root dev dependency

JorgenVatle committed:
Update vitest

JorgenVatle committed:
Update repository lockfile

JorgenVatle committed:
Update Vue plugin to latest

JorgenVatle committed:
Use meteor-vite@alpha

JorgenVatle committed:
Update vite-plugin-solid to v2.11.0

JorgenVatle committed:
Update Vite react plugin to v4.3.4

JorgenVatle committed:
Update svelte example to Svelte v5 and Vite v6

github-actions[bot] committed:
Version Packages (alpha)

github-actions[bot] committed:
RELEASING: Releasing 2 package(s)

Releases:
  jorgenvatle_vite@1.0.0-alpha.3
  meteor-vite@3.0.0-alpha.4

[skip ci]

JorgenVatle committed:
Merge pull request #249 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Correct initial version for jorgenvatle:vite package

JorgenVatle committed:
Merge remote-tracking branch 'origin/v3.1-vite-vms' into v3.1-vite-vms

JorgenVatle committed:
docs(changeset): Correct pre-release tag

JorgenVatle committed:
Use non-pre-release channel for jorgenvatle:vite

JorgenVatle committed:
Update changelog entry for jorgenvatle:vite

JorgenVatle committed:
docs(changeset): Use non-pre-release tag

JorgenVatle committed:
re-add missing initial version

JorgenVatle committed:
Bump initial version to 1.0.1

JorgenVatle committed:
Remove redundant changeset

JorgenVatle committed:
Update current version tag in jorgenvatle_vite package

JorgenVatle committed:
Update changelog entry for new release tag

github-actions[bot] committed:
Version Packages (alpha)

github-actions[bot] committed:
RELEASING: Releasing 1 package(s)

Releases:
  jorgenvatle_vite@1.1.0-alpha.0

[skip ci]

JorgenVatle committed:
Merge pull request #251 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Remove outdated svelte-meteor-data package

JorgenVatle committed:
Remove svelte-meteor-data from dependencies

JorgenVatle committed:
Add Svelte useTracker utility

JorgenVatle committed:
Use local useTracker implementation

JorgenVatle committed:
Correct mounting process for Svelte app

JorgenVatle committed:
Add useSubscribe utility

JorgenVatle committed:
Add debug log to useSubscribe tracker

JorgenVatle committed:
Use readable store for useTracker

JorgenVatle committed:
Refactor Svelte app to use local modernized Svelte tracker library

JorgenVatle committed:
Correct runtime subscription

JorgenVatle committed:
Use named export for RuntimeCollection

JorgenVatle committed:
Check whether react preamble should be included in markup

JorgenVatle committed:
Add react preamble to projects that need it

JorgenVatle committed:
Wrap React preamble in script tag

JorgenVatle committed:
Remove react-refresh preamble from example project now that it's provided automatically by Meteor-Vite.

JorgenVatle committed:
Update vite package

JorgenVatle committed:
Update completed/redundant roadmap entries

JorgenVatle committed:
Update installation instructions to match new package name

JorgenVatle committed:
Remove old 'vite-bundler' package in favor of rewritten 'vite' package.

JorgenVatle committed:
Update/remove references to old vite-bundler package

JorgenVatle committed:
Remove include path for deleted vite-bundler package

JorgenVatle committed:
Remove lockfile from meteor-vite package

JorgenVatle committed:
Remove lockfile from zodern-relay package

JorgenVatle committed:
Remove readme section about minifier packages.

They are still necessary to bundle Meteor individual packages into a single browser-friendly module. Without the minifier packages, browsers will have to pull in non-minified sources for all installed Meteor packages, which can result in a lot of unnecessary requests.

JorgenVatle committed:
Disable caching for 404 responses

JorgenVatle committed:
Fix type error for input file

JorgenVatle committed:
Use module format for server output

JorgenVatle committed:
Build for node21

JorgenVatle committed:
Use original file path for server entries

JorgenVatle committed:
Disable keepNames option as it appears to cause issues with downstream Meteor builds

JorgenVatle committed:
docs(changeset): Disable 'keepNames' feature from ESBuild - it appears to cause issues with downstream Meteor builds

github-actions[bot] committed:
Version Packages (alpha)

github-actions[bot] committed:
RELEASING: Releasing 2 package(s)

Releases:
  meteor-vite@3.0.0-alpha.5
  jorgenvatle_vite@1.1.0-alpha.1

[skip ci]

JorgenVatle committed:
Merge pull request #252 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Avoid use of arrow functions for meteorInstall to preserve 'this' context

JorgenVatle committed:
Externalize Meteor imports for server builds

JorgenVatle committed:
Disable meteor stubs plugin when building for the server

JorgenVatle committed:
docs(changeset): Disable Meteor package stubs when building for the server. This addresses some strange context issues with packages like redis-oplog when using stubs in intermediary server server bundle. The stubs are really only necessary in the server runtime environment, so this should be the safest way to build anyway.

JorgenVatle committed:
Revert "Avoid use of arrow functions for meteorInstall to preserve 'this' context"

This reverts commit 0e7ea7acf797efe0af42d5a2094da0fd135b6b5c.

github-actions[bot] committed:
Version Packages (alpha)

github-actions[bot] committed:
RELEASING: Releasing 1 package(s)

Releases:
  meteor-vite@3.0.0-alpha.6

[skip ci]

JorgenVatle committed:
Merge pull request #253 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Add todo for EJSON types

JorgenVatle committed:
Merge remote-tracking branch 'origin/v3.1-vite-vms' into v3.1-vite-vms

JorgenVatle committed:
Update call order for 404 middleware

JorgenVatle committed:
Remove unused type import

JorgenVatle committed:
docs(changeset): Fix call order for production Vite asset 404 middleware

github-actions[bot] committed:
Version Packages (alpha)

github-actions[bot] committed:
RELEASING: Releasing 1 package(s)

Releases:
  meteor-vite@3.0.0-alpha.7

[skip ci]

JorgenVatle committed:
Merge pull request #254 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Set plaintext content type for 404 page.

JorgenVatle committed:
docs(changeset): Hotfix WebApp handler response call order and set an explicit content-type header for invalid Vite asset requests

JorgenVatle committed:
Merge remote-tracking branch 'origin/v3.1-vite-vms' into v3.1-vite-vms

JorgenVatle committed:
Remove deleted vite-bundler from workspaces

github-actions[bot] committed:
RELEASING: Releasing 1 package(s)

Releases:
  meteor-vite@3.0.0-alpha.8

[skip ci]

github-actions[bot] committed:
Version Packages (alpha)

JorgenVatle committed:
Merge pull request #255 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Include hash in entry modules

JorgenVatle committed:
docs(changeset): Include hash in entry module filenames. Addresses an issue where older entry modules would be kept indefinitely by client and edge cached.

JorgenVatle committed:
Update wording for changelog entry

JorgenVatle committed:
Merge remote-tracking branch 'origin/v3.1-vite-vms' into v3.1-vite-vms

github-actions[bot] committed:
RELEASING: Releasing 1 package(s)

Releases:
  meteor-vite@3.0.0-alpha.9

[skip ci]

github-actions[bot] committed:
Version Packages (alpha)

JorgenVatle committed:
Merge pull request #256 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages (alpha)

JorgenVatle committed:
Update pre-release version

JorgenVatle committed:
Merge branch 'release' into merge-release-into-v4

# Conflicts:
#	.bin/release-meteor-package.mjs
#	.changeset/config.json
#	.changeset/pre.json
#	.github/workflows/release.yml
#	.gitignore
#	README.md
#	examples/react/.meteor/packages
#	examples/react/.meteor/versions
#	examples/react/package-lock.json
#	examples/react/package.json
#	examples/solid/.meteor/packages
#	examples/solid/.meteor/versions
#	examples/solid/package-lock.json
#	examples/solid/package.json
#	examples/svelte/.meteor/packages
#	examples/svelte/.meteor/versions
#	examples/svelte/package-lock.json
#	examples/svelte/package.json
#	examples/vue-ssr/.meteor/packages
#	examples/vue-ssr/.meteor/versions
#	examples/vue-ssr/package-lock.json
#	examples/vue-ssr/package.json
#	examples/vue-zodern-relay/.meteor/packages
#	examples/vue-zodern-relay/.meteor/versions
#	examples/vue-zodern-relay/package-lock.json
#	examples/vue-zodern-relay/package.json
#	examples/vue/.meteor/packages
#	examples/vue/.meteor/versions
#	examples/vue/package-lock.json
#	examples/vue/package.json
#	npm-packages/meteor-vite/CHANGELOG.md
#	npm-packages/meteor-vite/package.json
#	npm-packages/meteor-vite/src/meteor/IPC/methods/index.ts
#	npm-packages/meteor-vite/src/meteor/ServerBuilder.ts
#	package-lock.json
#	packages/tsconfig.json
#	packages/vite-bundler/CHANGELOG.md
#	packages/vite-bundler/package.js
#	packages/vite-bundler/src/build.ts
#	packages/vite-bundler/src/loading/vite-connection-handler.ts
#	packages/vite-bundler/src/server.ts
#	packages/vite-bundler/src/utility/Helpers.ts
#	packages/vite-bundler/src/vite-boilerplate/development.ts
#	packages/vite-bundler/src/workers.ts
#	packages/vite/package.json

JorgenVatle committed:
Remove changes to removed vite-bundler package

JorgenVatle committed:
Remove --force flag from npm link call

JorgenVatle committed:
Remove debugging middleware from production environment

JorgenVatle committed:
Update peer dependency to require full Vite v6 release

JorgenVatle committed:
Merge pull request #260 from JorgenVatle/v4

Merge release branch into Meteor-Vite v4 pre-release branch

JorgenVatle committed:
Add include reference for vite package index types

JorgenVatle committed:
Merge remote-tracking branch 'origin/v3.1-vite-vms' into v3.1-vite-vms

JorgenVatle committed:
Correct path to Meteor package

JorgenVatle committed:
Update installation instructions to include sections for both Meteor v2 and Meteor v3

JorgenVatle committed:
Add release branch link to readme for Meteor v2

JorgenVatle committed:
Update order of release branch link

JorgenVatle committed:
Revert "Update order of release branch link"

This reverts commit 8dcffc1a92787be361df84586f51179a1b5837af.

JorgenVatle committed:
Place release branch link in footnote

JorgenVatle committed:
Update initial v4 changelog entry to use a major release and provide additional details and migration steps

JorgenVatle committed:
Update initial version for meteor-vite to use full 2.0 release

JorgenVatle committed:
Require at least Node v20 for meteor-vite

JorgenVatle committed:
Remove meteor-v2 example app

JorgenVatle committed:
Exit pre-release mode

JorgenVatle committed:
Update repository lockfile

JorgenVatle committed:
Reformat major changelog entry

JorgenVatle committed:
Update initial version for jorgenvatle:vite package

JorgenVatle committed:
Update heading for changelog entry

JorgenVatle committed:
Increase heading level for major changes

JorgenVatle committed:
Update version tag for vite package

JorgenVatle committed:
Add separate section for server HMR in changelog

JorgenVatle committed:
Increase heading size for changelog summary

JorgenVatle committed:
Update repository lockfile

JorgenVatle committed:
Update migration steps to include install command for meteor-vite v3

JorgenVatle committed:
Use explicit version for jorgenvatle:vite release

JorgenVatle committed:
Revert "Use explicit version for jorgenvatle:vite release"

This reverts commit c6d347be18061addef79d0331245291557974871.

JorgenVatle committed:
Update react-meteor-data import path in readme

github-actions[bot] committed:
RELEASING: Releasing 3 package(s)

Releases:
  meteor-vite@3.0.0
  jorgenvatle_vite@1.0.0
  @meteor-vite/plugin-zodern-relay@1.1.0

[skip ci]

github-actions[bot] committed:
Version Packages

JorgenVatle committed:
Merge pull request #261 from JorgenVatle/changeset-release/v3.1-vite-vms

Version Packages
